src/ui/public/vislib/visualizations/marker_types/scaled_circles.js

```diff
@@ -13,45 +16,35 @@ export default function ScaledCircleMarkerFactory(Private) {
    * @param params {Object}
    */
   _.class(ScaledCircleMarker).inherits(BaseMarker);
+
   function ScaledCircleMarker(map, geoJson, params) {
-    let self = this;
+
     ScaledCircleMarker.Super.apply(this, arguments);
 
-    // multiplier to reduce size of all circles
-    let scaleFactor = 0.6;
+    let maxGeohashPrecision = 1;
+    for (let i = 0; i < geoJson.features.length; i += 1) {
+      maxGeohashPrecision = Math.max(geoJson.features[i].properties.geohash.length, maxGeohashPrecision);
+    }
+    const worldInPixels = 256 * Math.pow(2, this.map.getZoom());//map is 256 pixels wide at zoom level 0
+    const geoHashWidthInPixels = worldInPixels / geohashCells(maxGeohashPrecision, 0);
+    const geoHashHeightInPixels = worldInPixels / geohashCells(maxGeohashPrecision, 1);
+    const maxRadius = Math.max(geoHashWidthInPixels, geoHashHeightInPixels) / 2;//bias the largest dimension, so we get some overlap
+    const minRadius = maxRadius / 3;//magic number to make sure smallest circle is not too small.
```

Comment:
I get that. But why 3? Would 4 work? How about 10? There must be a reason 3 is used here, and some context in the form of a variable name and some comments would help explain that.